CMakeLists.txt

```diff
@@ -147,12 +103,33 @@ install(
   COMPONENT freeciv21)
 
 if(WIN32 OR MSYS OR MINGW)
-  # Install the libraries compiled in the section above
-  install(
-    PROGRAMS
-    ${CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS}
-    DESTINATION ${CMAKE_INSTALL_BINDIR}
-    COMPONENT freeciv21)
+  install(CODE [[
+    message(STATUS "Collecting dependencies for freeciv21 executables...")
+
+    # Take a variable that is available at "install" time and repurpose
+    string(REGEX REPLACE "objdump.exe" "" MINGW_PATH ${CMAKE_OBJDUMP})
+
+    # Function to analyze the third party dll files linked to the exe's
+    #   Uses the repurposed variable from above to tell the function where
+    #   the dll files are located. Ignores dll's that come with Windows.
+    file(GET_RUNTIME_DEPENDENCIES
+      RESOLVED_DEPENDENCIES_VAR r_deps
+      UNRESOLVED_DEPENDENCIES_VAR u_deps
+      DIRECTORIES ${MINGW_PATH}
+      PRE_EXCLUDE_REGEXES "^api-ms-*"
+      POST_EXCLUDE_REGEXES "C:[\\\\/][Ww][Ii][Nn][Dd][Oo][Ww][Ss][\\\\/].*"
+      EXECUTABLES
+        "${CMAKE_INSTALL_PREFIX}/freeciv21-*.exe"
+      )
+      message(STATUS "Installing library dependencies for freeciv21 executables...")
+      foreach(_file ${r_deps})
+        file(INSTALL
+          DESTINATION "${CMAKE_INSTALL_PREFIX}"
+          TYPE SHARED_LIBRARY
+          FILES "${_file}"
```

Comment:
Can't we avoid the loop and pass `${r_deps}` directly here? `FILES` makes me think that this should be possible.

Reply:
I got error messages when doing this so went with the loop to correct

Reply:
Fair enough

Reply:
second commit forthcoming...